### PR TITLE
[CI] Adopt GitHub actions

### PR DESCRIPTION
**Pull request by FranzBusch**
# Motivation

We want to migrate all of repositories to use GitHub Actions.

# Modification

This PR adopts GH actions in this repo.

# Result

New and shiny CI.

### CODE REVIEW

**Review by Lukasa — APPROVED** [review]

**Comment by FranzBusch** [comment]
@Lukasa @simonjbeaumont this PR is ready. I have split this into multiple commits to make it easier to review

File: CONTRIBUTORS.txt
Comment (simonjbeaumont) [inline CONTRIBUTORS.txt:1]: Why was this renamed and do we need to worry about anyone having linked to this file?

**Review by simonjbeaumont — APPROVED** [review]

File: CONTRIBUTORS.txt
Comment (FranzBusch) [reply CONTRIBUTORS.txt:1]: This was renamed to align it with the majority of our other repos. There was a weird inconsistency that I wasn't aware of before. I am not worried that anybody linked to it.

**Review by Lukasa — APPROVED** [review]
